ui/src/variables/utils/hydrateVars.ts

```diff
@@ -126,9 +126,10 @@ const findSubgraph = (
   variables: Variable[]
 ): VariableNode[] => {
   const subgraph: Set<VariableNode> = new Set()
-
   // use an ID array to reduce the chance of reference errors
   const varIDs = variables.map(v => v.id)
+  // create an array of IDs to reference later
+  const graphIDs = []
```

Comment:
nit for Ari: this should be an object for a faster lookup on line 148, but i can't imagine the search space being very large, so it's not a blocker